### ISSUE TITLE
App startup crashes on iOS 17.0 / NewsstandKit issues

### ISSUE THREAD

**Issue by dalexsoto** [xamarin/xamarin-macios#18964]
If you have a Xamarin.Forms / MAUI app that targets iOS and has been built with Xcode 14.3.1 there is a high chance that this app will crash at startup on iOS 17. This is because [Apple removed NewsstandKit](https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-17-release-notes#Newsstand) from iOS 17.0 and there is a chance that your App is referencing this Framework depending on your linker settings.

In order to avoid this problem you will need to recompile your application:

### .NET / MAUI

* Make sure you are using at least Visual Studio 17.7.3+ which includes iOS workload 16.4.7098/7.0.100

### Xamarin.iOS / Xamarin.Forms

* Make sure you are using at least [xamarin.ios-16.4.0.18.pkg](https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg) / [xamarin.mac-9.3.0.18.pkg](https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/ad75896c1d3c57af522e9802c1d477a3/xamarin.mac-9.3.0.18.pkg) in your Mac build host. 

## Detect if an app is affected

It's possible to detect whether an app is affected by using `otool` from the command line to check if the app is linking (not weakly) with the NewsstandKit framework.

This app is affected, because `cmd=LC_LOAD_DYLIB`:

```shell
$ otool -l path/to/Debug/testapp.app/testapp | grep NewsstandKit -C 3
Load command 19
          cmd LC_LOAD_DYLIB
      cmdsize 88
         name /System/Library/Frameworks/NewsstandKit.framework/NewsstandKit (offset 24)
   time stamp 2 Thu Jan  1 01:00:02 1970
      current version 1.0.0
compatibility version 1.0.0
```

This app is _not_ affected, because `cmd=LC_LOAD_WEAK_DYLIB` means it's linking weakly with `NewsstandKit`:

```shell
$ otool -l path/to/Debug/testapp.app/testapp | grep NewsstandKit -C 3
Load command 19
          cmd LC_LOAD_WEAK_DYLIB
      cmdsize 88
         name /System/Library/Frameworks/NewsstandKit.framework/NewsstandKit (offset 24)
   time stamp 2 Thu Jan  1 01:00:02 1970
      current version 1.0.0
compatibility version 1.0.0
```

This app is  also _not_ affected, because it doesn't link with NewsstandKit at all:

```shell
$ otool -l path/to/Debug/testapp.app/testapp | grep NewsstandKit -C 3
[no output]
```

Alternatively install an iOS 17 beta on a device and try to run the app. If it launches successfully, it's not affected.

**Comment by RobSchoenaker** [comment]
The Visual Studio versions is correct? This version is not available for Mac OS. Latest version for Max OS is 17.6.3

**Comment by mlancione** [comment]
Does this only affect Xamarin.Forms and MAUI apps or also Xamarin.iOS apps?

**Comment by RobSchoenaker** [comment]
FWIW: VS on Mac will ask for upgrade after manually installing the 16.4.018 package. Do not install that update, since is it a downgrade and will cause your iOS app to crash again.

**Comment by mlancione** [comment]
Is there a way to debug Xamarin.iOS apps on iOS 17? The debugger doesn't seem to connect for me despite launching the app manually. Does this require XCode 15 and therefore will not be supported for Xamarin.iOS apps?

**Comment by mlancione** [comment]
> > Does this require XCode 15 and therefore will not be supported for Xamarin.iOS apps?
> 
> I am using XCode 14.3.1, since Xamarin Forms does not support XCode 15. I can debug apps but had a hard time pushing a new version to Apple for review. Not sure why your debug session fails. Did you remove the app from the development device completely?

Debugging started working after a reboot. Debugging also seems to work using XCode 15 as well.

Although not officially supported by Microsoft, does the fact that Xamarin.iOS 16.4.0.18 seem to work with XCode 15 beta 8 to build a Xamarin.iOS app mean that we may have an additional year of use out of it before having to switch to .Net? This would be assuming we don't need iOS 17 specific APIs and assuming no breaking changes by Apple with future XCode 15 point releases.

**Comment by RobSchoenaker** [comment]
Gogo to hear that a reboot worked.

WOW! If XCode 15 'works' with Xamarin.iOS that would be very good news indeed. There was another issue raised here with a lot op people complaining that XCode 15 is not supported - and never will, unless the source code is ported to MAUI.

Even if this is not officially supported, it would give everyone some slack.

**Comment by jtorvald** [comment]
@RobSchoenaker  That would make me and my team very happy indeed. 

I can add to this that the additional build argument ` --weak-framework=NewsstandKit.framework/NewsstandKit` as suggested by @RobSchoenaker in a previous issue also helps to work around this issue. We have multiple apps in production that needed this to be able to work in iOS17. This will result in NewsstandKit `LC_LOAD_WEAK_DYLIB`.

**Comment by RobSchoenaker** [comment]
@jtorvald - I may be repeating something here, but just to be clear: you can omit that build argument when using the linked "xamarin.ios-16.4.0.18.pkg" package from @dalexsoto 
Just do NOT update VS for Mac when prompted after installing it!

**Comment by jtorvald** [comment]
@RobSchoenaker good to know. Can I ask what happens if we would update VS for Mac at this point?
I'm running `17.6.3 (build 421)`

**Comment by RobSchoenaker** [comment]
@jtorvald I will try to clarify - probably was not clear enough.

@dalexsoto mentioned the update package in the first message in this thread. When you install that package on your Mac and (re)start Visual Studio, you will be prompted to install an older version of xamarin.ios. I believe it was 16.4.0.16, which *obviously* will cause debugging to fail again if you removed the additional build argument.

Just ignore the message from VS for Mac to 'update', which is actually a 'downgrade'. VS for Mac will need to be updated to mention the 'update/downgrade' now. 

Does this make sense?

**Comment by jtorvald** [comment]
@RobSchoenaker yeah that makes sense. I thought you wrote not to update VS itself. Thanks.

**Comment by phillippschmedt** [comment]
@dalexsoto Hi, you are referring to [https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg](url) - Where did you find this release and where can I see the release notes? Thanks a lot

**Comment by LeadAssimilator** [comment]
I'm confused...I've updated VS2022 to 17.7.4 which includes XMI 16.4.0.15 and I let it update the paired mac to the same 16.4.0.15.  I'm no longer seeing strong references to NewsstandKit like I was previously, but weak references and I didn't change any mtouch args to force it as weak either.  As such, installing 16.4.0.18 appears to be entirely unnecessary.

So what is 16.4.0.18 is really need for then if the strong reference is gone in 16.4.0.15?

**Comment by rolfbjarne** [comment]
> I'm confused...I've updated VS2022 to 17.7.4 which includes XMI 16.4.0.15 and I let it update the paired mac to the same 16.4.0.15. I'm no longer seeing strong references to NewsstandKit like I was previously, but weak references and I didn't change any mtouch args to force it as weak either. As such, installing 16.4.0.18 appears to be entirely unnecessary.
> 
> So what is 16.4.0.18 is really need for then if the strong reference is gone in 16.4.0.15?

We had to do multiple fixes to remove the strong reference in all cases.

Your project may only run into the first scenarios we fixed (and released) - which means you don't need 16.4.0.18, 16.4.0.15 may be enough. That doesn't mean other's won't.

**Comment by LeadAssimilator** [comment]
Ok thanks.  It would be great if these versions were tagged to make tracking easier...

For anyone that is curious, the 16.4 packages are built from the [xcode14.3](/xamarin/xamarin-macios/tree/xcode14.3) branch and .15 has hash 8c3876d5e and .18 has hash 9d266025e.  Here is the compare https://github.com/xamarin/xamarin-macios/compare/8c3876d5e...9d266025e.

To find the hashes for a pkg you can cat /Library/Frameworks/Xamarin.iOS.framework/Versions/w.x.y.z/buildinfo if you install it, or extract it via pkg-util on the mac or 7zip from windows - it is nested inside the monotouch.pkg/Payload/Payload~ file.

**Comment by IainS1986** [comment]
Has the Azure XamariniOS Pipeline task (https://moasure.visualstudio.com/Moasure2020/_build/results?buildId=1408&view=logs&j=d0c55d57-dc89-5219-b43a-87fc3bd7617e&t=760a608d-127a-5ce2-ccd3-71f0e97908ad&l=7) Have the correct above mentioned Xamarin IOS package update?

Can we rely on build from Azure using this task to work fine on iOS 17?

**Comment by IainS1986** [comment]
By the looks of the log, its only using 16.0.0.72 with XCode 14.2 - which seems quite far behind.

Is there anything we can do to ensure the latest is used? Are there plan to update Azure devops (and soon)?

I see the MacOS image that was used in one of our builds is macos-12 ... so this might be on us for neglecting to use a newer image (I can see macOS-13 is available, but that macOS-latest still points to 12?)
https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software

**Comment by rolfbjarne** [comment]
> By the looks of the log, its only using 16.0.0.72 with XCode 14.2 - which seems quite far behind.
> 
> Is there anything we can do to ensure the latest is used? Are there plan to update Azure devops (and soon)?
> 
> I see the MacOS image that was used in one of our builds is macos-12 ... so this might be on us for neglecting to use a newer image (I can see macOS-13 is available, but that macOS-latest still points to 12?) [learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software)
> 
> EDIT: Oh. Great. MacOS-13 image doesn't have Xamarin installed/support?
> 
> So is Azure DevOps support for Xamarin dropped sooner than the May 2024 deadlline?

I think these questions are better asked to the Azure DevOps team here: https://github.com/actions/runner-images/issues

**Comment by IainS1986** [comment]
> Good to hear that a reboot worked.
> 
> WOW! If XCode 15 'works' with Xamarin.iOS that would be very good news indeed. There was another issue raised here with a lot op people complaining that XCode 15 is not supported - and never will, unless the source code is ported to MAUI.
> 
> Even if this is not officially supported, it would give everyone some slack.

Can Verify. I've installed 16.4.0.18 on my Mac, XCode 15 (not the beta, the release) and just ran a debug build and debugger connected all fine to my iPhone.

I don't have the NewsstandKit API issue as were Xamarin native with linker turned on so it is/was being stripped anyway.

But I have Xamarin IOS, on a mac, building with Xcode 15 and debugging straight to my iOS 17 device 👍 
(Note with iOS17 I *do* have to tap to launch the app)


(No idea if an IPA would go through app store submittion mind)

**Comment by bobwhitten** [comment]
I'm a bit confused on how install Xamarin.iOS16.4.0.18 properly.  I use VS for Windows.  After installing the package on my Mac, of course now when I pair, VS complains that the versions are mismatched.
I feel like I missed somewhere the proper way to install the package or reconfig on VS for Windows.

AND, note -- I installed XCode 15 before this (somewhat unintentionally), and I can still build & debug on my iOS16.6 devices, but not on my iOS 17 devices.

**Comment by svaldetero** [comment]
> For macOS you can just install the package "xamarin.ios-16.4.0.18.pkg" (link in the description)

How do I get this to work for a Maui / Microsoft.iOS project on macOS? I have 7.0.401 & 8.0.100-rc.1 installed and I still get the error. I tried adding `<MtouchExtraArgs>--weak-framework=NewsstandKit.framework/NewsstandKit</MtouchExtraArgs>` and still get the error.

**Comment by rolfbjarne** [comment]
> I'm a bit confused on how install Xamarin.iOS16.4.0.18 properly. I use VS for Windows. After installing the package on my Mac, of course now when I pair, VS complains that the versions are mismatched.
I feel like I missed somewhere the proper way to install the package or reconfig on VS for Windows.

Windows is a bit more complicated, but I believe there's a way to skip the version check. I'll investigate and get back to you.

**Comment by rolfbjarne** [comment]
> > For macOS you can just install the package "xamarin.ios-16.4.0.18.pkg" (link in the description)
> 
> How do I get this to work for a Maui / Microsoft.iOS project on macOS? I have 7.0.401 & 8.0.100-rc.1 installed and I still get the error. I tried adding `<MtouchExtraArgs>--weak-framework=NewsstandKit.framework/NewsstandKit</MtouchExtraArgs>` and still get the error.

What's the output of `dotnet workload list` and `dotnet --info`?

**Comment by SamiAlan** [comment]
I'm developing on Windows as well and got the following output:
`C:\Users\sami>dotnet workload list

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
maui-android               7.0.92/7.0.100         VS 17.7.34031.279
android                    33.0.68/7.0.100        VS 17.7.34031.279
maccatalyst                16.4.7098/7.0.100      VS 17.7.34031.279
ios                        16.4.7098/7.0.100      VS 17.7.34031.279
maui-windows               7.0.92/7.0.100         VS 17.7.34031.279
maui-maccatalyst           7.0.92/7.0.100         VS 17.7.34031.279
maui-ios                   7.0.92/7.0.100         VS 17.7.34031.279

Use `dotnet workload search` to find additional workloads to install.


C:\Users\sami>dotnet --info
.NET SDK:
 Version:   7.0.401
 Commit:    eb26aacfec

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.401\

Host:
  Version:      7.0.11
  Architecture: x64
  Commit:       ecb34f85ec

.NET SDKs installed:
  7.0.400 [C:\Program Files\dotnet\sdk]
  7.0.401 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download`

**Comment by RagibNoor** [comment]
> Hi, you are referring to [https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg](url) - Where did you find this release and where can I see the release notes? Thanks a lot

I have the same question 
searched on the Microsoft page but could not find anything

**Comment by Hackmodford** [comment]
> @dalexsoto Hi, you are referring to [https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg](url) - Where did you find this release and where can I see the release notes? Thanks a lot

Where does one find these releases listed? I can only find older packages.

**Comment by rolfbjarne** [comment]
@SamiAlan can you get a [binlog](https://aka.ms/binlog) and upload it here?

**Comment by SamiAlan** [comment]
Sure. I cloned the project and removed the bits that are confidential. Here is the link: [https://drive.google.com/file/d/1NtRXeG5uedXp9vYwTCtRjRndPp7VOQr5/view?usp=sharing](url)

**Comment by kolodiichyk** [comment]
Hi everybody,
I think that faced with the same issue:

MT5209: Error: warning: ignoring duplicate libraries: '-lz'  Native linking
MT5209: Error: framework 'NewsstandKit' not found Native linking
MTOUCH: Error MT5202: Native linking failed. Please review the build log. (MT5202)
MTOUCH: Error MT5202: Native linking failed. Please review the build log. (MT5202) 

Just create a new Xamarin.iOS or Xamarin.Forms project and try to build. It's from MacBook macOS 13.5.2 (22G91) 


dotnet workload list:

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
wasm-tools                 7.0.11/7.0.100         SDK 7.0.300
macos                      13.3.7099/7.0.100      SDK 7.0.300
maui-maccatalyst           7.0.92/7.0.100         SDK 7.0.300
maui-ios                   7.0.92/7.0.100         SDK 7.0.300
maui-android               7.0.92/7.0.100         SDK 7.0.300

dotnet --info:

.NET SDK:
 Version:   7.0.308
 Commit:    f6a2f50f2d

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx.13-x64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.308/

Host:
  Version:      7.0.11
  Architecture: x64
  Commit:       ecb34f85ec

.NET SDKs installed:
  3.1.407 [/usr/local/share/dotnet/sdk]
  3.1.408 [/usr/local/share/dotnet/sdk]
  3.1.409 [/usr/local/share/dotnet/sdk]
  3.1.412 [/usr/local/share/dotnet/sdk]
  3.1.413 [/usr/local/share/dotnet/sdk]
  3.1.414 [/usr/local/share/dotnet/sdk]
  3.1.415 [/usr/local/share/dotnet/sdk]
  3.1.416 [/usr/local/share/dotnet/sdk]
  3.1.420 [/usr/local/share/dotnet/sdk]
  3.1.426 [/usr/local/share/dotnet/sdk]
  5.0.201 [/usr/local/share/dotnet/sdk]
  5.0.202 [/usr/local/share/dotnet/sdk]
  5.0.203 [/usr/local/share/dotnet/sdk]
  5.0.400 [/usr/local/share/dotnet/sdk]
  5.0.401 [/usr/local/share/dotnet/sdk]
  5.0.402 [/usr/local/share/dotnet/sdk]
  5.0.403 [/usr/local/share/dotnet/sdk]
  5.0.404 [/usr/local/share/dotnet/sdk]
  5.0.408 [/usr/local/share/dotnet/sdk]
  6.0.408 [/usr/local/share/dotnet/sdk]
  6.0.412 [/usr/local/share/dotnet/sdk]
  6.0.413 [/usr/local/share/dotnet/sdk]
  6.0.414 [/usr/local/share/dotnet/sdk]
  7.0.302 [/usr/local/share/dotnet/sdk]
  7.0.306 [/usr/local/share/dotnet/sdk]
  7.0.307 [/usr/local/share/dotnet/sdk]
  7.0.308 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.14 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.15 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.18 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.19 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.20 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.20 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.22 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.15 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.18 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.19 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

**Comment by IoTFier** [comment]
When pairing to my MAC, I'm getting following error with XCode 15 installed.

An error occurred while trying to start the Broker...
An error occurred while trying to start Broker 17.7.0.216

**Comment by rolfbjarne** [comment]
@SamiAlan 

> Sure. I cloned the project and removed the bits that are confidential. Here is the link: [https://drive.google.com/file/d/1NtRXeG5uedXp9vYwTCtRjRndPp7VOQr5/view?usp=sharing](url)

As far as I can tell that's a successful build?

**Comment by rolfbjarne** [comment]
@kolodiichyk 

> MT5209: Error: warning: ignoring duplicate libraries: '-lz' Native linking MT5209: Error: framework 'NewsstandKit' not found Native linking MTOUCH: Error MT5202: Native linking failed. Please review the build log. (MT5202) Error: linker command failed with exit code 1 (use -v to see invocation)
> 
> Just create a new Xamarin.iOS or Xamarin.Forms project and try to build. It's from MacBook macOS 13.5.2 (22G91)

Which version of Xamarin.iOS do you have?

You can get that using the command line:

```
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mtouch --version
```

**Comment by kolodiichyk** [comment]
mtouch 16.4.0.6 (xcode14.3: 97731c92c)

**Comment by rolfbjarne** [comment]
@kolodiichyk 

> mtouch 16.4.0.6 (xcode14.3: https://github.com/xamarin/xamarin-macios/commit/97731c92cc6d147825c1a39ed2c5c530a5f9a12b)

You need to install [xamarin.ios-16.4.0.18](https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg), like the description of this issue says.

**Comment by Eagle104** [comment]
I can build and the project works through Pair to Mac etc, But I am not able to debug my application anymore on iOS 17

**Comment by SamiAlan** [comment]
> @SamiAlan
> 
> > Sure. I cloned the project and removed the bits that are confidential. Here is the link: [https://drive.google.com/file/d/1NtRXeG5uedXp9vYwTCtRjRndPp7VOQr5/view?usp=sharing](url)
> 
> As far as I can tell that's a successful build?

Yes, The project is building successfully. It is also deploying the app to the IOS device but once I start debugging and open the app, it suddenly shuts down while at the splash screen. Just for clarity, I had a `MAX_PATH` issue earlier where I Installed a nuget package called `Plugin.Firebase.CloudMessaging`. I couldn't build the project and the error was that It could find part of the path to a file related to the package (even though I am sure it exists). I was able to resolve it by deleted the `_CopyLocalBindingResources` target in the `"C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7098\targets\Xamarin.Shared.Sdk.targets"` file. I'm not sure if that breaks the app now after the update

**Comment by rolfbjarne** [comment]
@Eagle104 

> I can build and the project works through Pair to Mac etc, But I am not able to debug my application anymore on iOS 17

Does the app launch/work correctly if you tap on it?

**Comment by rolfbjarne** [comment]
@SamiAlan
> > > Sure. I cloned the project and removed the bits that are confidential. Here is the link: [https://drive.google.com/file/d/1NtRXeG5uedXp9vYwTCtRjRndPp7VOQr5/view?usp=sharing](url)
> > 
> > 
> > As far as I can tell that's a successful build?
> 
> Yes, The project is building successfully. It is also deploying the app to the IOS device but once I start debugging and open the app, it suddenly shuts down while at the splash screen. Just for clarity, I had a `MAX_PATH` issue earlier where I Installed a nuget package called `Plugin.Firebase.CloudMessaging`. I couldn't build the project and the error was that It could not find part of the path to a file related to the package (even though I am sure it exists). I was able to resolve it by deleted the `_CopyLocalBindingResources` target in the `"C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7098\targets\Xamarin.Shared.Sdk.targets"` file. I'm not sure if that breaks the app now after the update

You can tell if it's something else or notx by following the "Detect if an app is affected" instructions in the description of this issue.

You could also try disabling Hot Restart to see if that makes a difference.

**Comment by rolfbjarne** [comment]
@Eagle104
> > > I can build and the project works through Pair to Mac etc, But I am not able to debug my application anymore on iOS 17
> > 
> > 
> > Does the app launch/work correctly if you tap on it?
> 
> VS Version 2022 Pro 17.6.5
> 
> Xamarin.Forms 5.0.0.2312
> 
> XCode 15
> 
> Xamarin.iOS 16.4.0.18 - Installed Manually
> 
> Mac M2 Ventura 13.5.2
> 
> Yes the application opens and Works.

OK, that's a different problem then, and I suggest you file a new issue for it.

**Comment by rolfbjarne** [comment]
@bobwhitten 

> > I'm a bit confused on how install Xamarin.iOS16.4.0.18 properly. I use VS for Windows. After installing the package on my Mac, of course now when I pair, VS complains that the versions are mismatched.
> > I feel like I missed somewhere the proper way to install the package or reconfig on VS for Windows.
> 
> Windows is a bit more complicated, but I believe there's a way to skip the version check. I'll investigate and get back to you.

In order to skip the version check you have to add/edit a registry key:

Create key named `DisableXamariniOSValidation` (DWORD) with the value 1 under: `Computer\HKEY_CURRENT_USER\Software\Xamarin\VisualStudio\MacAgent`

Remember to delete this key once the updated version has been released to Visual Studio as well.

**Comment by LeadAssimilator** [comment]
@rolfbjarne 

> @bobwhitten
> 
> > > I'm a bit confused on how install Xamarin.iOS16.4.0.18 properly. I use VS for Windows. After installing the package on my Mac, of course now when I pair, VS complains that the versions are mismatched.
> > > I feel like I missed somewhere the proper way to install the package or reconfig on VS for Windows.
> > 
> > 
> > Windows is a bit more complicated, but I believe there's a way to skip the version check. I'll investigate and get back to you.
> 
> In order to skip the version check you have to add/edit a registry key:
> 
> Create key named `DisableXamariniOSValidation` (DWORD) with the value 1 under: `Computer\HKEY_CURRENT_USER\Software\Xamarin\VisualStudio\MacAgent`
> 
> Remember to delete this key once the updated version has been released to Visual Studio as well.

Any chance we could get a checkbox for that in the VS UI under Xamarin -> iOS Settings?  Installs of newer packages could perhaps even always clear the flag too to serve as a reminder?  Where should I file that feature request?

**Comment by LeadAssimilator** [comment]
@rolfbjarne 

> > > > Windows is a bit more complicated, but I believe there's a way to skip the version check. I'll investigate and get back to you.
> > > 
> > > 
> > > In order to skip the version check you have to add/edit a registry key:
> > > Create key named `DisableXamariniOSValidation` (DWORD) with the value 1 under: `Computer\HKEY_CURRENT_USER\Software\Xamarin\VisualStudio\MacAgent`
> > > Remember to delete this key once the updated version has been released to Visual Studio as well.
> > 
> > 
> > Any chance we could get a checkbox for that in the VS UI under Xamarin -> iOS Settings? Installs of newer packages could perhaps even always clear the flag too to serve as a reminder? Where should I file that feature request?
> 
> Instructions for suggestions are here: https://learn.microsoft.com/en-us/visualstudio/ide/suggest-a-feature?view=vs-2022

Ok thanks.  I was expecting it to have to go somewhere else or there would be a more direct path to whomever controls the xamarin specific settings pages.

**Comment by RhomGit** [comment]
> @kolodiichyk
> 
> > mtouch 16.4.0.6 (xcode14.3: [97731c9](https://github.com/xamarin/xamarin-macios/commit/97731c92cc6d147825c1a39ed2c5c530a5f9a12b))
> 
> You need to install [xamarin.ios-16.4.0.18](https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg), like the description of this issue says.

Logging in to my Mac host and installing this pkg file fixed the problem. I can now compile and run on the simulator.
I haven't tried publishing to the App Store yet.

**Comment by pasha-o** [comment]
I tried the suggestions above using VS 2022 on Windows, and everything works on the simulator and app launches. but I can't test it on a real hardware device with iOS 17. it simply gives me the error below
error MT0000: Unexpected error - Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new		 	0

**Comment by fit4you-bike** [comment]
Hi everybody, I'm also having this error compiling against XCode 15 and 17 iOS sdk

clang : error : linker command failed with exit code 1 (use -v to see invocation)
error MT5209 : Native linking error : warning: ignoring duplicate libraries: '-lz'
error MT5209 : Native linking error : framework 'NewsstandKit' not found
MTOUCH : error MT5202: Native linking failed. Please review the build log.

visual studio for mac 17.6.4 (build 472)

Xamarin.iOS 16.4.0.18 (xcode14.3: 9d266025e) using framework: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk

xcode Version 15.0 (15A240d)

.NET SDK:
 Version:   7.0.401
 Commit:    eb26aacfec

Ambiente di runtime:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx.13-x64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.401/

Host:
  Version:      7.0.11
  Architecture: x64
  Commit:       ecb34f85ec


I'm unable to link besides the fact I've hadded 

--weak-framework=NewsstandKit.framework/NewsstandKit

to mtouch options. Any ideas? Thanks in advance

**Comment by stoff99** [comment]
> > Good to hear that a reboot worked.
> > WOW! If XCode 15 'works' with Xamarin.iOS that would be very good news indeed. There was another issue raised here with a lot op people complaining that XCode 15 is not supported - and never will, unless the source code is ported to MAUI.
> > Even if this is not officially supported, it would give everyone some slack.
> 
> Can Verify. I've installed 16.4.0.18 on my Mac, XCode 15 (not the beta, the release) and just ran a debug build and debugger connected all fine to my iPhone.
> 
> I don't have the NewsstandKit API issue as were Xamarin native with linker turned on so it is/was being stripped anyway.
> 
> But I have Xamarin IOS, on a mac, building with Xcode 15 and debugging straight to my iOS 17 device 👍 (Note with iOS17 I _do_ have to tap to launch the app)
> 
> (No idea if an IPA would go through app store submittion mind)
> 
> EDIT - Also this VS Mac only, Rider doesn't appear to recognise the iOS 17 device when its plugged in like VS Mac does. But I managed to build to an iOS 16 device with XCode 15 in Rider fine.

@IainS1986 please can you vote for the Rider ticket regarding they do not display iOS 17 real devices please. Ticket: https://youtrack.jetbrains.com/issue/RIDER-99240/Deploy-Debug-to-iOS-17-devices-via-XCode-15-using-Xamarin-16.4.0.18

**Comment by SamiAlan** [comment]
@rolfbjarne 

> @SamiAlan
> 
> > > > Sure. I cloned the project and removed the bits that are confidential. Here is the link: [https://drive.google.com/file/d/1NtRXeG5uedXp9vYwTCtRjRndPp7VOQr5/view?usp=sharing](url)
> > > 
> > > 
> > > As far as I can tell that's a successful build?
> > 
> > 
> > Yes, The project is building successfully. It is also deploying the app to the IOS device but once I start debugging and open the app, it suddenly shuts down while at the splash screen. Just for clarity, I had a `MAX_PATH` issue earlier where I Installed a nuget package called `Plugin.Firebase.CloudMessaging`. I couldn't build the project and the error was that It could not find part of the path to a file related to the package (even though I am sure it exists). I was able to resolve it by deleted the `_CopyLocalBindingResources` target in the `"C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7098\targets\Xamarin.Shared.Sdk.targets"` file. I'm not sure if that breaks the app now after the update
> 
> You can tell if it's something else or not by following the "Detect if an app is affected" instructions in the description of this issue.
> 
> You could also try disabling Hot Restart to see if that makes a difference.

`otool` only works on MacOS. Is there an alternative to check if there is a hard linked with `NewsstandKit ` ?

**Comment by jonasmayor** [comment]
Hello all with I managed to compile and run the application with the following configuration ( on Iphone and Ipad)
Visual studio for mac 17.6.4 (build 472)
Xamarin.iOS 16.4.0.18 
Xcode Version 15.0 (15A240d)

But there is a problem that neither CollectionView not ListView work , the app crashes with an Object C error 
like error  UIGraphicsBeginImageContext deprecated when using UIGraphicsContext , use UIGraphicsImageRenderer instead

Any idea where that could happend in the implementation of the CollectionView or the ListView?  I'm using Datatemplates to render my Items Collections 
Thanks

**Comment by velocitysystems** [comment]
@dalexsoto Thanks for this helpful post. We can confirm building with Xcode 15 and Xamarin.iOS 16.4.0.18 works. Just wondering though if this is a supported scenario from Microsoft or is it a "use at your own risk" approach? Thanks!

**Comment by exendahal** [comment]
My app is working well with iPadOS 17.0.1 but crashing with iOS 17. 
using `--weak-framework=NewsstandKit.framework/NewsstandKit` doesn't seem to have any improvement.

**Comment by RagibNoor** [comment]
I am having a strange issue  
My app is working on the 17.0 simulator but it's not working on the actual device iPad 6th Gen ios 17.0
In My app, I am using Xamarin Forms web view 
Build Env  :
Xcode 15 and Xamarin.iOS 16.4.0.18
VS for mac 17.6.4(build 472)
**otool is not showing any output** 

here is the build configuration 
```
<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|iPhone' ">
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\iPhone\Debug</OutputPath>
    <DefineConstants>DEBUG</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <MtouchArch>ARM64</MtouchArch>
    <CodesignKey> *********</CodesignKey>
    <MtouchDebug>true</MtouchDebug>
    <CodesignEntitlements>Entitlements.plist</CodesignEntitlements>
    <CodesignExtraArgs />
    <CodesignResourceRules />
    <MtouchLink>None</MtouchLink>
    <MtouchExtraArgs>-v -v -v -v --dlsym:Microsoft.Win32.Registry.dll</MtouchExtraArgs>
    <DeviceSpecificBuild>false</DeviceSpecificBuild>
    <MtouchHttpClientHandler>NSUrlSessionHandler</MtouchHttpClientHandler>
    <OptimizePNGs>false</OptimizePNGs>
    <Nullable>
    </Nullable>
    <CodesignProvision>********</CodesignProvision>
    <BuildIpa>true</BuildIpa>
    <IpaMetadata>Info.plist</IpaMetadata>
    <IpaPackageName>xamarin-dev-usv</IpaPackageName>
  </PropertyGroup>
```

```
<PropertyGroup Condition=” ‘$(Configuration)|$(Platform)’ == ‘Debug|iPhoneSimulator’ “>
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\iPhoneSimulator\Debug</OutputPath>
    <DefineConstants>DEBUG</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <MtouchArch>x86_64</MtouchArch>
    <MtouchLink>None</MtouchLink>
    <MtouchDebug>true</MtouchDebug>
    <CodesignKey>*********</CodesignKey>
    <MtouchHttpClientHandler>HttpClientHandler</MtouchHttpClientHandler>
    <DeviceSpecificBuild>false</DeviceSpecificBuild>
    <MtouchExtraArgs>-v -v -v -v --dlsym:Microsoft.Win32.Registry.dll</MtouchExtraArgs>
    <CodesignEntitlements>Info.plist</CodesignEntitlements>
    <OptimizePNGs>false</OptimizePNGs>
    <ExternalConsole>false</ExternalConsole>
    <MtouchUseLlvm>
    </MtouchUseLlvm>
    <CodesignProvision>*********</CodesignProvision>
  </PropertyGroup>
```

**Comment by exendahal** [comment]
I was able to fix the issue with `--weak-framework=NewsstandKit.framework/NewsstandKit` and send an update to the app store.

**Comment by jonathan-snd** [comment]
@dalexsoto Thank you, I confirm that building with Xcode 15 and Xamarin.iOS 16.4.0.18 works

**Comment by Eagle104** [comment]
Are you still able to debug your app as well with Xcode 15 and Xamarin.iOS 16.4.0.18. I am building through Pair to Mac. Everything works but the app does not launch at the end to allow for debugging.

> @dalexsoto Thank you, I confirm that building with Xcode 15 and Xamarin.iOS 16.4.0.18 works

**Comment by jonathan-snd** [comment]
> Are you still able to debug your app as well with Xcode 15 and Xamarin.iOS 16.4.0.18. I am building through Pair to Mac. Everything works but the app does not launch at the end to allow for debugging.
> 
> > @dalexsoto Thank you, I confirm that building with Xcode 15 and Xamarin.iOS 16.4.0.18 works

After installing the application bundle, visual studio remains in the "waiting for the debugger to connect to ..." state.
to solve I launch the application manually in the device, that way I can debbugging with visual studio

**Comment by robertmaxted** [comment]
I can confirm that it all works as expected using the following:

Xcode: 15 (15A240d).
Xamarin.iOS version: xamarin.ios-16.4.0.18.pkg - (https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg).
Visual studio Mac 2022: 17.6.4 (build 472).
iOS (iPhone) version: 17.0.1.

**Comment by SamiAlan** [comment]
Using Windows here and I can also confirm that the project is built and deployed successfully on my physical IPhone device.

Visual Studio: 17.7.4
Xamarin.iOS: 16.4.7098
IPhone: 17.0.1

**Comment by dev2442422** [comment]
@dalexsoto @rolfbjarne - are you able to comment on if building Xamarin apps against Xcode 15 and Xamarin.iOS 16.4.0.18 will work to meet the April 2024 Xcode 15 build requirement? I am having issues migrating to .NET Maui, and hoping to buy some time before the deadline. Thanks!

**Comment by dev2442422** [comment]
@dalexsoto  @rolfbjarne  are you able to comment on if building Xamarin apps against Xcode 15 and Xamarin.iOS 16.4.0.18 will work to meet the April 2024 Xcode 15 build requirement? I am having challenges migrating to .NET Maui, and hoping to buy some time before the deadline. Thanks!

**Comment by IoTFier** [comment]
> Using Windows here and I can also confirm that the project is built and deployed successfully on my physical IPhone device.
> 
> Visual Studio: 17.7.4 Xamarin.iOS: 16.4.7098 IPhone: 17.0.1

Hi @SamiAlan, 

VS 17.7.4 is throwing following error when connecting to my MAC -
"The Xamarin.iOS SDK version '16.4.0.6' that is installed on the Mac is not compatible with this version of Visual Studio. Would you like us to install Xamarin.iOS '16.4.0.15' for you? This will overwrite any other existing Xamarin.iOS installation on your Mac."

Can you please confirm the version of Xamarin.iOS on your Mac? I just got the 16.4.0.6 as I upgraded the VS on Mac.

Thank you

**Comment by bobwhitten** [comment]
> > Using Windows here and I can also confirm that the project is built and deployed successfully on my physical IPhone device.
> > Visual Studio: 17.7.4 Xamarin.iOS: 16.4.7098 IPhone: 17.0.1
> 
> Hi @SamiAlan,
> 
> VS 17.7.4 is throwing following error when connecting to my MAC - "The Xamarin.iOS SDK version '16.4.0.6' that is installed on the Mac is not compatible with this version of Visual Studio. Would you like us to install Xamarin.iOS '16.4.0.15' for you? This will overwrite any other existing Xamarin.iOS installation on your Mac."
> 
> Can you please confirm the version of Xamarin.iOS on your Mac? I just got the 16.4.0.6 as I upgraded the VS on Mac.
> 
> Thank you

Did you see this further back -- it worked for me in getting rid of that error, and I'm able to build. 
In order to skip the version check you have to add/edit a registry key:

Create key named DisableXamariniOSValidation (DWORD) with the value 1 under: Computer\HKEY_CURRENT_USER\Software\Xamarin\VisualStudio\MacAgent

Remember to delete this key once the updated version has been released to Visual Studio as well.

**Comment by shawyunz** [comment]
> Hello all with I managed to compile and run the application with the following configuration ( on Iphone and Ipad) Visual studio for mac 17.6.4 (build 472) Xamarin.iOS 16.4.0.18 Xcode Version 15.0 (15A240d)
> 
> But there is a problem that neither CollectionView not ListView work , the app crashes with an Object C error like error UIGraphicsBeginImageContext deprecated when using UIGraphicsContext , use UIGraphicsImageRenderer instead
> 
> Any idea where that could happend in the implementation of the CollectionView or the ListView? I'm using Datatemplates to render my Items Collections Thanks
> 
> PS: I guess this should be a new thread but almost one one is paying attention to the new bugs that are surging with Xamarin Forms and IOS 17

@jonasmayor Have you solved your error yet?

**Comment by quintonv** [comment]
> ould go through app



> I can build and the project works through Pair to Mac etc, But I am not able to debug my application anymore on iOS 17

Hi How did you get your windows paired machine to build? I am trying but getting so many issues with ios 17 connecting to my mac with xcode 15. Any advice would be greatly appreciated

**Comment by brandonDevMobile** [comment]
Im using visual studio 17.6.4 on a mac with a xamarin forms project (not maui) and when I update to xcode 15 cant see any simulators for ios 17 or any other version. Any Help please?

xamarin ios version: 16.4.0.6
xamarin mac version:  9.3.0.6

**Comment by IainS1986** [comment]
> xamarin ios version: 16.4.0.6

I'd read the first comment in this thread.

**Comment by shawyunz** [comment]
> Im using visual studio 17.6.4 on a mac with a xamarin forms project (not maui) and when I update to xcode 15 cant see any simulators for ios 17 or any other version. Any Help please?
> 
> xamarin ios version: 16.4.0.6 xamarin mac version: 9.3.0.6

It's in the top thread, [xamarin.ios-16.4.0.18.pkg](https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg)

**Comment by krassx** [comment]
Adding my two cents here too. I've tried the proposed workaround with installing ```Xamarin.iOS 16.4.0.18.pkg```. While I can see the list of devices and simulators, I still can't deploy and debug application. The build/deploy/run process fails with the following:

```
error HE0004: Could not load the framework 'IDEProducts' (path: /Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/IDEProducts): 
dlopen(/Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/IDEProducts, 0x0001): Library not loaded: @rpath/IDEDistribution.framework/Versions/A/IDEDistribution
  Referenced from: <1C42486B-CF36-3FFB-A22C-F084A9B30A6B> /Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/Versions/A/IDEProducts
  Reason: tried: '/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.60/tools/lib/mlaunch/mlaunch.app/Contents/Frameworks/IDEDistribution.framework/Versions/A/IDEDistribution' (no such file), '/Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/Versions/A/Frameworks/IDEDistribution.framework/Versions/A/IDEDistribution' (no such file), '/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.60/tools/lib/mlaunch/mlaunch.app/Contents/Frameworks/IDEDistribution.framework/Versions/A/IDEDistribution' (no such file), '/Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/Versions/A/Frameworks/IDEDistribution.framework/Versions/A/IDEDistribution' (no such file)
```

My environment:
```
Darwin 23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:34 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T8103 arm64
```

```
Visual Studio Community 2022 for Mac
Version 17.6.4 (build 472)
Installation UUID: 12504c59-951e-4fbf-b686-f8d1e7c0cef1

Runtime
.NET 7.0.3 (64-bit)
Architecture: Arm64
Microsoft.macOS.Sdk 13.1.1007; git-rev-head:8afca776a0a96613dfb7200e0917bb57f9ed5583; git-branch:release/7.0.1xx-xcode14.2

Roslyn (Language Service)
4.6.0-3.23180.6+99e956e42697a6dd886d1e12478ea2b27cceacfa

NuGet
Version: 6.4.0.117

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.308/Sdks
SDK Versions:
	7.0.308
	7.0.307
	7.0.306
	7.0.304
	7.0.302
	7.0.203
	7.0.202
	7.0.201
	7.0.103
	7.0.101
	7.0.100
	6.0.414
	6.0.413
	6.0.412
	6.0.410
	6.0.408
	6.0.407
	6.0.406
	6.0.404
	6.0.403
	6.0.402
	6.0.400
	6.0.302
	6.0.301
	6.0.300
	6.0.202
	6.0.200
	6.0.101
	6.0.100
MSBuild SDKs: /Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET SDK (x64)
SDK Versions:
	6.0.414
	6.0.413
	6.0.412
	6.0.410
	6.0.408
	6.0.407
	6.0.406
	6.0.404
	6.0.403
	6.0.402
	6.0.400
	6.0.302
	6.0.301
	6.0.300
	6.0.202
	6.0.103
	6.0.102
	6.0.101
	5.0.408
	5.0.407
	5.0.406
	5.0.405
	5.0.404
	3.1.426
	3.1.425
	3.1.424
	3.1.422
	3.1.421
	3.1.420
	3.1.419
	3.1.418
	3.1.417
	3.1.416

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	7.0.11
	7.0.10
	7.0.9
	7.0.7
	7.0.5
	7.0.4
	7.0.3
	7.0.1
	7.0.0
	6.0.22
	6.0.21
	6.0.20
	6.0.18
	6.0.16
	6.0.15
	6.0.14
	6.0.12
	6.0.11
	6.0.10
	6.0.8
	6.0.7
	6.0.6
	6.0.5
	6.0.4
	6.0.2
	6.0.1
	6.0.0

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	6.0.22
	6.0.21
	6.0.20
	6.0.18
	6.0.16
	6.0.15
	6.0.14
	6.0.12
	6.0.11
	6.0.10
	6.0.8
	6.0.7
	6.0.6
	6.0.5
	6.0.4
	6.0.3
	6.0.2
	6.0.1
	5.0.17
	5.0.16
	5.0.15
	5.0.14
	5.0.13
	3.1.32
	3.1.31
	3.1.30
	3.1.28
	3.1.27
	3.1.26
	3.1.25
	3.1.24
	3.1.23
	3.1.22

Xamarin.Profiler
Version: 1.8.0.49
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode: 15.0 22265
Build: 15A240d

Xamarin.Mac
Version: 9.3.0.18 Visual Studio Community
Hash: 9d266025e
Branch: xcode14.3
Build date: 2023-09-06 19:52:26-0400

Xamarin.iOS
Version: 16.4.0.18 Visual Studio Community
Hash: 9d266025e
Branch: xcode14.3
Build date: 2023-09-06 19:52:27-0400

Xamarin.Android
Version: 13.2.1.2 (Visual Studio Community)
Commit: xamarin-android/d17-5/a8a26c7
Android SDK: /Users/akarimov/Library/Android/sdk
	Supported Android versions:
		12.1 (API level 32)
		12.0 (API level 31)
		8.1  (API level 27)
		11.0 (API level 30)
		10.0 (API level 29)
		7.1  (API level 25)
		9.0  (API level 28)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 34.0.4
SDK Build Tools Version: 33.0.2

Build Information: 
Mono: d9a6e87
Java.Interop: xamarin/java.interop/d17-5@149d70fe
SQLite: xamarin/sqlite/3.40.1@68c69d8
Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-5@ca1552d

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.16.1
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.6.0.50
Hash: a715dca
Branch: HEAD
Build date: 2023-09-07 02:05:26 UTC

Android Device Manager
Version: 0.0.0.1309
Hash: 06e3e77
Branch: HEAD
Build date: 2023-09-07 02:05:26 UTC

Xamarin Designer
Version: 17.6.3.9
Hash: 2648399ae8
Branch: remotes/origin/d17-6
Build date: 2023-09-07 02:05:20 UTC

Build Information
Release ID: 1706040472
Git revision: 0b8c2cb9f01ef14a2b07ff4ea047268c8756fee6
Build date: 2023-09-07 02:03:50+00
Build branch: release-17.6
Build lane: release-17.6

Operating System
Mac OS X 14.0.0
Darwin 23.0.0 Darwin Kernel Version 23.0.0
    Fri Sep 15 14:41:34 PDT 2023
    root:xnu-10002.1.13~1/RELEASE_ARM64_T8103 arm64
```

**Comment by keozx** [comment]
same as above, but I have .NET 8 RC

**Comment by pvma** [comment]
> I can confirm that it all works as expected using the following:
> 
> Xcode: 15 (15A240d). Xamarin.iOS version: xamarin.ios-16.4.0.18.pkg - (https://download.visualstudio.microsoft.com/download/pr/de7f727f-243c-4429-8773-eebd735eb1bb/da4a65f27ace7f805914ef63661e0efd/xamarin.ios-16.4.0.18.pkg). Visual studio Mac 2022: 17.6.4 (build 472). iOS (iPhone) version: 17.0.1.

This also worked for me. Plus, tried the same setup with Visual Studio for Mac, and also worked

**Comment by phillippschmedt** [comment]
It works fine for me too with this version and XCode 15. Only issue is I need to manually start the app when debugging. 

My main question that remains is, are we safe for iOS 17.x in general now? So does this mean we have until iOS 18/Xcode 16 before we can run into issues again?

**Comment by MariusPCK** [comment]
> It works fine for me too with this version and XCode 15. Only issue is I need to manually start the app when debugging.
> 
> My main question that remains is, are we safe for iOS 17.x in general now? So does this mean we have until iOS 18/Xcode 16 before we can run into issues again?

You are never safe my friend, this is Microsoft. No but on a serious note.. You are bretty much fucked for the future unless you move to MAUI :o

**Comment by krassx** [comment]
@pvma @phillippschmedt Are you on a macOS Sonoma or some other version? For me, the proposed workaround only fixes the absence of devices/simulators in the list of deployement targets. Application itself is not deployed to the selected device.

**Comment by pvma** [comment]
> @pvma @phillippschmedt Are you on a macOS Sonoma or some other version? For me, the proposed workaround only fixes the absence of devices/simulators in the list of deployement targets. Application itself is not deployed to the selected device.

Yes, I am on macOS Sonoma

**Comment by phillippschmedt** [comment]
https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-rc-2/

Bonus 2: Xcode 15 and Android API 34
Xamarin developers can now use Xcode 15 to target the latest versions (e.g iOS 17, iPad 17), and build for Android API 34 in order to be compliant with store policies. To do this, install Visual Studio 17.8 Preview 3 or the latest stable version of Visual Studio for Mac and configure your environment as usual. This does not provide newer platform APIs, but does enable existing projects to continue building while you complete your upgrades to .NET 8 and .NET MAUI regardless of the [Xamarin end-of-support date next year](https://dotnet.microsoft.com/platform/support/policy/xamarin).

**Comment by mnottbohm** [comment]
I updated today to the newest VS Version 17.7.5, but still getting the error on XCODE 15 and ISO 17.0.3

The application get pushed to the device, but debugger crashing with:

Rebooted everything already many time!

error MT1007: Failed to launch the application '/Users/markus/Library/Caches/Xamarin/mtbs/builds/CycleProCut/fddf4f2e7c374baf78f3512d57eb3202069ef7c9fd586893e5d728a34a19d339/bin/Debug/net7.0-ios/ios-arm64/device-builds/iphone12.5-17.0.3/CycleProCut.app' on the device 'iPhone 11 von Markus': Failed to launch the application 'com.companyname.cycleprocut' on the device 'iPhone 11 von Markus': Invalid Service Error (error: 0xe8000022)

Thanks,

Markus

**Comment by rolfbjarne** [comment]
We've released the fix for this:

* https://github.com/xamarin/xamarin-macios/releases/tag/dotnet-8.0.1xx-rc2-8968
* https://github.com/xamarin/xamarin-macios/releases/tag/xamarin-ios-16.4.0.18
* https://github.com/xamarin/xamarin-macios/releases/tag/dotnet-7.0.3xx-7107

> App doesn't auto start after deployment, I need to manually start it on the device.

This is a known issue, Apple completely rewrote how the communication between the device and the Mac happens, so we have to figure out how to support launching on device using their new code. Hopefully we'll have this fixed relatively soon.

**Comment by svaldetero** [comment]
> This is a known issue, Apple completely rewrote how the communication between the device and the Mac happens, so we have to figure out how to support launching on device using their new code.

Is there a Github item we can follow that tracks that issue?

**Comment by rolfbjarne** [comment]
> > This is a known issue, Apple completely rewrote how the communication between the device and the Mac happens, so we have to figure out how to support launching on device using their new code.
> 
> Is there a Github item we can follow that tracks that issue?

I've implemented a fix, and we'll try to get this released asap.

**Comment by thedee** [comment]
> > > This is a known issue, Apple completely rewrote how the communication between the device and the Mac happens, so we have to figure out how to support launching on device using their new code.
> > 
> > 
> > Is there a Github item we can follow that tracks that issue?
> 
> I've implemented a fix, and we'll try to get this released asap.

Anymore updates on this? We have customers bombing and need a fix asap.   Our Xamarin.Form apps crashes right aways as explained in this post.

**Comment by IainS1986** [comment]
It's explained in the first post how to fix it.

Needing to launch the app manually to debug is a separate thing - and does not stop you doing a build the will work for your customers.

**Comment by thedee** [comment]
> It's explained in the first post how to fix it.
> 
> Needing to launch the app manually to debug is a separate thing - and does not stop you doing a build the will work for your customers.

We make our build with a DevOps Azure pipleline so we are kindof stuck.

**Comment by noahcolvin** [comment]
> We make our build with a DevOps Azure pipleline so we are kindof stuck.

We use VS App Center, which is very slow to update, but the `weak-framework` fix has worked and has been deployed for weeks already without issue.

**Comment by IainS1986** [comment]
Well you're in for bad news, this fix is unlikely to make it onto azure DevOps. I posted asking and basically got a 'no' response.

This issue is, it needs osx13 in order to install the later Xcode, and the mac13 image does not have Xamarin support. Only the lower mac12 image does, but it can't have Xcode 15 so can't have this fix installed.

Your options are to do manual builds and upload those.

Or you can target mac13 image but you'll have to have build steps to install the Xamarin packages you need first.

Either way you'll likely have to look into sorting this yourself I think.

**Comment by thedee** [comment]
> > We make our build with a DevOps Azure pipleline so we are kindof stuck.
> 
> We use VS App Center, which is very slow to update, but the `weak-framework` fix has worked and has been deployed for weeks already without issue.

With the "--weak-framework=NewsstandKit.framework/NewsstandKit" args are you able to debug or just make a build and it works?   I tried the args and debugging doesn't work but have yet to try making an ipa and testing that out.

**Comment by noahcolvin** [comment]
> > > We make our build with a DevOps Azure pipleline so we are kindof stuck.
> > 
> > 
> > We use VS App Center, which is very slow to update, but the `weak-framework` fix has worked and has been deployed for weeks already without issue.
> 
> With the "--weak-framework=NewsstandKit.framework/NewsstandKit" args are you able to debug or just make a build and it works? I tried the args and debugging doesn't work but have yet to try making an ipa and testing that out.

Yes, I haven't updated Xcode or installed the latest VS and Xamarin packages (so that I match App Center) and can still debug locally. I added that argument to all the build configs and it's working fine in iOS 17. App Center had no issue building and uploading to the store. Had it released a week before iOS 17 dropped.

**Comment by beeradmoore** [comment]
People keep mentioning AppCenter so I'll put this bit of info here.

Use boots to install later version Xamarin iOS and Android packages and don't wait for them to update it.

https://github.com/jonathanpeppers/boots

Additionally I went back and forwards with AppCenter support about Xcode 14.3 support. It (and 15) are available in some of their other iOS build systems (native and react from memory) but not the Xamarin ones. 

I asked if we they can just add the build step to install Xamarin with booths on the later macOS images which should then have all the other required parts.

That has gone back to their engineering team to determine if it's possible/viable 🤞

**Comment by thedee** [comment]
As it turns out I didn't have to do the --weak args.  All I did was to add this to the App.xaml:

```
<Application ...
             xmlns:ios="clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core"
             ios:Application.HandleControlUpdatesOnMainThread="true">
    ...
</Application>

```

One of the crashes we were only gettiing on ios17 was:
```
Message=Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread.
StackTrace:
  at ObjCRuntime.Runtime.ThrowNSException (System.IntPtr ns_exception) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/16.4.0.18/src/Xamarin.iOS/ObjCRuntime/Runtime.cs:459 
  at ObjCRuntime.Runtime.throw_ns_exception (System.IntPtr exc) [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/runtime/Delegates.generated.cs:277 
  at (wrapper native-to-managed) ObjCRuntime.Runtime.throw_ns_exception(intptr)
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend(intptr,intptr)
  at CoreAnimation.CATransaction.Commit () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/16.4.0.18/src/Xamarin.iOS/CoreAnimation/CATransaction.g.cs:90 
  at Xamarin.Forms.Platform.iOS.CADisplayLinkTicker.StartThread () [0x0003a] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CADisplayLinkTicker.cs:61 
  at (wrapper delegate-invoke) <Module>.invoke_void()
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/threading/thread.cs:74 
  at (wrapper delegate-invoke) <Module>.invoke_void_object(object)
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:968 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:910 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:899 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/threading/thread.cs:111 
  at (wrapper delegate-invoke) <Module>.invoke_void()
```

After doing that my application currently is working fine.  We are still doing more test but ios17 so far starts up with no problems.  Hopefully customers will have same good results.
